src/test/java/ro/developmentfactory/thedrones/service/DronePositionServiceImplTest.java

```diff
@@ -0,0 +1,267 @@
+
+        // Then
+        assertEquals(expectedX, response.getCurrentPositionX());
+        assertEquals(expectedY, response.getCurrentPositionY());
+        assertEquals(direction, response.getFacingDirection());
+        verify(droneStatusRepository).save(droneStatus);
+    }
+
+    @ParameterizedTest
+    @EnumSource(Direction.class)
+    @DisplayName("Turning Left: Valid turn should update direction correctly")
+    void turningLeft_ValidTurn_ShouldUpdateDirection(Direction initialDirection) {
+        // Given
+        Direction expectedDirection = initialDirection.turnLeft();
+        droneStatus.setFacingDirection(initialDirection);
+        when(droneRepository.findById(any(UUID.class))).thenReturn(Optional.of(drone));
+
+        // When
+        DroneStatusResponse response = dronePositionService.turn(drone.getIdDrone(), TurnDirection.LEFT);
+
+        // Then
+        assertEquals(expectedDirection, response.getFacingDirection());
+        verify(droneStatusRepository).save(droneStatus);
+    }
+
+    @Test
+    @DisplayName("Move Forward: Count move should be updated correctly")
+    void moveForward_CountMove_ShouldIncrement() {
+        // Given
+        when(droneRepository.findById(any(UUID.class))).thenReturn(Optional.of(drone));
+
+        // When
+        dronePositionService.moveForward(drone.getIdDrone());
+
+        // Then
+        assertEquals(1, drone.getCountMove());
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "N, E, RIGHT",
+            "E, S, RIGHT",
+            "S, W, RIGHT",
+            "W, N, RIGHT",
+            "N, W, LEFT",
+            "W, S, LEFT",
+            "S, E, LEFT",
+            "E, N, LEFT"
+    })
+    @DisplayName("Turning Direction: Rotation should update direction correctly")
+    void turningDirection_ValidRotation_ShouldUpdateDirectionCorrectly(Direction initialDirection, Direction expectedDirection, TurnDirection turnDirection) {
+        // Given
+        droneStatus.setFacingDirection(initialDirection);
+        when(droneRepository.findById(any(UUID.class))).thenReturn(Optional.of(drone));
+
+        // When
+        DroneStatusResponse response = dronePositionService.turn(drone.getIdDrone(), turnDirection);
+
+        // Then
+        assertEquals(expectedDirection, response.getFacingDirection());
+        verify(droneStatusRepository).save(droneStatus);
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "0,0,N,0,1",
+            "0,0,E,1,0",
+            "0,0,S,0,-1",
+            "0,0,W,-1,0",
+            "1,1,N,1,2",
+            "1,1,E,2,1",
+            "1,1,S,1,0",
+            "1,1,W,0,1",
+            "0,9,N,0,10",
+            "9,0,E,10,0",
+            "9,9,S,9,8",
+            "9,9,W,8,9"
+    })
+    @DisplayName("Move Forward: Position should be updated correctly based on direction")
+    void moveForward_PositionShouldBeUpdatedCorrectly(int startX, int startY, Direction direction, int expectedX, int expectedY) {
```

Comment:
it's a bit hard to understand what you want to do in this case because you are mixing valid moves and invalid. having try-catch here is confusing. I would recommend splitting it into 2 tests: valid tests and those that need to throw `IllegalArgumentException`. for valid you already have `moveForward_ValidMoves_ShouldUpdatePositionCorrectly`, maybe you can combine it with other valid moves. then you can always assume to see the exception for invalid ones